### DISCUSSION TITLE
Generate SW multiple times warning...

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#39628] [Help] [answered]
I am trying to create a PWA  and I got the pages to work on a run build without any errors... however... I am getting this warning from the local host 3000

`warn - GenerateSW has been called multiple times, perhaps due to running webpack in --watch mode. The precache manifest generated after the first call may be inaccurate! Please see https://github.com/GoogleChrome/workbox/issues/1790 for more information.`

Would anybody know how to fix that?

Also.... I don't think I got the PWA working properly, because I can't see a "download" button of the page in the browser...

Why do you think I would be getting that error?   ... and what else would I need to convert it into a PWA?

The site I am working with is this one:
https://github.com/Elindo586/nextprojectb/tree/main/pages

**Comment by Elindo586** [comment]
So... I don't see this in my vscode public file:
https://yourdomain.com/sw.js and https://yourdomain.com/workbox-*.js.

Shouldn't those be generated automatically when I do npm run build? 


I can see everything was build OK supposedly? 

```

info  - Loaded env from C:\Users\Edgar\Desktop\Stuff\Acode\nextprojectb\.env.local
info  - Checking validity of types
> [PWA] Compile client (static)
> [PWA] Auto register service worker with: C:\Users\Edgar\Desktop\Stuff\Acode\nextprojectb\node_modules\next-pwa\register.js
> [PWA] Service worker: C:\Users\Edgar\Desktop\Stuff\Acode\nextprojectb\.next\sw.js
> [PWA]   url: /sw.js
> [PWA]   scope: /
> [PWA] Compile server
> [PWA] Compile server
info  - Creating an optimized production build  
info  - Compiled successfully
info  - Collecting page data  
info  - Generating static pages (29/29)
info  - Finalizing page optimization  



```

**Comment by Elindo586** [comment]
BTW... I am using this video as reference
https://www.youtube.com/watch?v=ARNN_zmrwcw&list=PLBqKyyql_CfaHCt156qaptkdPwhIYWMkD&index=7

^^ on this video the sw.js is generated during the build....   I also didn't create a server... 

Along with this link here... which I will add more meta data later...
https://www.npmjs.com/package/next-pwa

**Comment by gaston-flores** [comment]
This is more of an issue with `next-pwa` than Next. Check out https://github.com/shadowwalker/next-pwa/issues/371

**Comment by Elindo586** [comment]
![Capture33](https://user-images.githubusercontent.com/93617433/184736183-8831ef1e-1ca8-4963-b217-e1698b030c91.PNG)


I think it might have to do with the manifest as well....

**Comment by mashaole** [comment]
> I am trying to create a PWA and I got the pages to work on a run build without any errors... however... I am getting this warning from the local host 3000
> 
> `warn - GenerateSW has been called multiple times, perhaps due to running webpack in --watch mode. The precache manifest generated after the first call may be inaccurate! Please see https://github.com/GoogleChrome/workbox/issues/1790 for more information.`
> 
> Would anybody know how to fix that?
> 
> Also.... I don't think I got the PWA working properly, because I can't see a "download" button of the page in the browser...
> 
> Why do you think I would be getting that error? ... and what else would I need to convert it into a PWA?
> 
> The site I am working with is this one: https://github.com/Elindo586/nextprojectb/tree/main/pages

use this rather in next.config.js

const withPWA = require("next-pwa")({
  dest: "public",
});

module.exports = withPWA({
//your other existing next.config.js
});

**Comment by Elindo586** [comment, marked as answer]
I would suggest not using next-pwa any more.


My solution was to put the manifest.json  along with all the icon sizes you want, and the sw.js in the public foldder...  

 then created an app.js for service workers in the public folder......

then inside my _app.js in the pages folder called the manifest and the service workers like this:

```
<Head>
          <link rel="manifest" href="/manifest.json" />
other head stuff here...
<Head/>
```

then on the bottom of the _app.js page called the service workers like this:
`<script async type="module" src="app.js"></script>`

and it worked...

Gotta do a few more little additions/fixes for the PWA but it is working...

You can dig the code out in here:
https://github.com/Elindo586/nextprojectb

**Reply by anishFE** [reply]
is this valid for next js 15 + ?

**Reply by helderpgoncalves** [reply]
> is this valid for next js 15 + ?

Looking for an awnser also.

**Comment by richardebrain** [comment]
> I am trying to create a PWA and I got the pages to work on a run build without any errors... however... I am getting this warning from the local host 3000
> 
> `warn - GenerateSW has been called multiple times, perhaps due to running webpack in --watch mode. The precache manifest generated after the first call may be inaccurate! Please see https://github.com/GoogleChrome/workbox/issues/1790 for more information.`
> 
> Would anybody know how to fix that?
> 
> Also.... I don't think I got the PWA working properly, because I can't see a "download" button of the page in the browser...
> 
> Why do you think I would be getting that error? ... and what else would I need to convert it into a PWA?
> 
> The site I am working with is this one: https://github.com/Elindo586/nextprojectb/tree/main/pages

you need to disable pwa during development , doing that fix the error GenerateSw
